Adding support for seed based colour generation

MaciekBaron opened the pull request:
As discussed earlier I've introduced support for generating colours based on a seed.

A seed is a string of characters you can pass to the `make_colour()` function; the same string of characters will always return the same colour.

Example:

```
// Will always return "#bf8172"
Please.make_color({seed: 'foobar'});
// Will always return ["#72bf9c", "#9572bf", "#bf9972"]
Please.make_color({seed: 'GitHub', colors_returned: 3});
```

Note that if the colour generation code changes in a future version of the library, the colours generated for specific seeds might change (depends on the number of times the random functions are called).

MaciekBaron committed:
Adding support for seed based colour generation

teliosdev commented on Please.js:
```diff
@@ -171,12 +171,20 @@
-			return Math.random() * ( max - min ) + min;
+		function random_float( min, max, randomiser ){
+			var random = Math.random;
+			if (randomiser instanceof RC4Random) {
```
Wouldn't it be better to just check if `randomiser.random` was a function?

MaciekBaron replied:
No, because that would fail if `randomiser` was `undefined`, `null` or anything else other than a `function` or an `object`.

teliosdev replied:
Then check to see if `randomiser` is a function or an object.

MaciekBaron replied:
I don't understand how is that better or more efficient than checking if `randomiser` is an instance of `RC4Random`, which is guaranteed to have a `random` function?

teliosdev replied:
Because if we want to allow the user to give a random function later, then all we have to do is change a few lines down in `make_color`.

MaciekBaron replied:
To be honest I don't really see the point of having more than these 2 types of random functions, which fit all our needs. This isn't a cryptography library so "perfect" randomization is unnecessary, and that's the only use I can think of really. Unless you have something else in mind?

teliosdev replied:
Maybe I'm attempting to over generalize it, but allowing the user to create a "random" function that is used by the library seems like a benefit.

MaciekBaron replied:
If someone finds a use case that requires additional custom randomisation functions, we can implement it, it will be easy to do. At the moment, like I've said, I can't see any use of having another custom "random" function.

Fooidge replied:
Yeah it's likely overboard to have custom implementations of random. I think at the point which you'd seriously care about that you'd be using a more generalist library and generating your own colors.

MaciekBaron commented:
Are you going to merge soon? It already has merge conflicts...

Commit:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	Please.js

MaciekBaron commented:
I have fixed the merge conflict, but please merge soon so it doesn't happen again.

Commit:
Properly fixing conflict (forgot to save before)

Fooidge commented:
As soon as I can. I have to do code review and testing first and I've been really busy with my own features + other projects. Sorry it keeps slipping but I wouldn't expect it before the weekend. 

Jordan Checkman

> On Jul 16, 2014, at 5:26 AM, Maciek Baron notifications@github.com wrote:
> 
> I have fixed the merge conflict, but please merge soon so it doesn't happen again.
> 
> —
> Reply to this email directly or view it on GitHub.

MaciekBaron commented:
It's been two months now. I might just create a fork of your project and maintain it...

Fooidge commented:
Seriously dude?

I did mention that I moved cities and changed jobs right. I just got Internet this week.

Jordan Checkman

> On Sep 15, 2014, at 5:23 AM, Maciek Baron notifications@github.com wrote:
> 
> It's been two months now. I might just create a fork of your project and maintain it...
> 
> —
> Reply to this email directly or view it on GitHub.

MaciekBaron commented:
You never mentioned this in this thread. After seeing some commit activity
one can get the impression that this request has been ignored. If you're
still planning to have a look at it then that's great.
On 15 Sep 2014 14:05, "Jordan Checkman" notifications@github.com wrote:

> Seriously dude?
> 
> I did mention that I moved cities and changed jobs right. I just got
> Internet this week.
> 
> Jordan Checkman
> 
> > On Sep 15, 2014, at 5:23 AM, Maciek Baron notifications@github.com
> > wrote:
> > 
> > It's been two months now. I might just create a fork of your project and
> > maintain it...
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Fooidge/PleaseJS/pull/5#issuecomment-55587065.
